account_reversal/account_reversal.py

```diff
@@ -59,63 +56,58 @@ def _move_reversal(self, cr, uid, move, reversal_date,
 
         :return: Returns the id of the created reversal move
         """
-        if context is None:
-            context = {}
-        move_line_obj = self.pool.get('account.move.line')
-        period_obj = self.pool.get('account.period')
-        period_ctx = context.copy()
-        period_ctx['company_id'] = move.company_id.id
-        period_ctx['account_period_prefer_normal'] = True
+        period_obj = self.env['account.period']
 
         if not reversal_period_id:
-            reversal_period_id = period_obj.find(
-                cr, uid, reversal_date, context=period_ctx)[0]
+            reversal_period_id = period_obj.with_context(
+                company_id=self.company_id.id,
+                account_period_prefer_normal=True).find(reversal_date)[0]
         if not reversal_journal_id:
-            reversal_journal_id = move.journal_id.id
-
-        reversal_ref = ''.join([x for x in [move_prefix, move.ref] if x])
-        reversal_move_id = self.copy(cr, uid, move.id,
-                                     default={
-                                         'date': reversal_date,
-                                         'period_id': reversal_period_id,
-                                         'ref': reversal_ref,
-                                         'journal_id': reversal_journal_id,
-                                         'to_be_reversed': False,
-                                     },
-                                     context=context)
-
-        self.write(cr, uid, [move.id],
-                   {'reversal_id': reversal_move_id,
-                    # ensure to_be_reversed is true if ever it was not
-                    'to_be_reversed': True},
-                   context=context)
-
-        reversal_move = self.browse(cr, uid, reversal_move_id, context=context)
+            reversal_journal_id = self.journal_id.id
+
+        if self.env['account.journal'].browse([
+                reversal_journal_id]).company_id != self.company_id:
+            raise Warning('Wrong company Journal is %s but we have %s' % (
+                reversal_journal_id.company_id.name, self.company_id.name))
+        if reversal_period_id.company_id != self.company_id:
+            raise Warning('Wrong company Period is %s but we have %s' % (
+                reversal_journal_id.company_id.name, self.company_id.name))
+
+        reversal_ref = ''.join([x for x in [move_prefix, self.ref] if x])
+        reversal_move = self.copy(default={
+            'company_id': self.company_id.id,  # TODO SERIOUS ODOO BUG I'D SAY
```

Comment:
It's been a while, but I believe I found that the company_id was not being copied when using self.copy.